contracts/zero-ex/contracts/test/foundry/utils/ForkUtils.sol

```diff
@@ -0,0 +1,336 @@
+    }
+
+    function readLiquiditySourceAddresses() public returns (string memory) {
+        string memory root = vm.projectRoot();
+        string memory path = string(
+            abi.encodePacked(root, "/", "contracts/test/foundry/addresses/sourceAddresses.json")
+        );
+        sourcesJson = vm.readFile(path);
+        return vm.readFile(path);
+    }
+
+    function getLiquiditySourceAddresses(uint index) public returns (LiquiditySources memory sources) {
+        readLiquiditySourceAddresses();
+        bytes memory liquiditySources = sourcesJson.parseRaw(indexChainIds[index]);
+        return abi.decode(liquiditySources, (LiquiditySources));
+    }
+
+    function readAddresses() public returns (string memory) {
+        string memory root = vm.projectRoot();
+        string memory path = string(abi.encodePacked(root, "/", "contracts/test/foundry/addresses/addresses.json"));
+        addressesJson = vm.readFile(path);
+        return vm.readFile(path);
+    }
+
+    function getContractAddresses(uint index) public returns (Addresses memory addresses) {
+        readAddresses();
+        bytes memory contractAddresses = addressesJson.parseRaw(indexChainIds[index]);
+        return abi.decode(contractAddresses, (Addresses));
+        //log_named_address("WETH/NATIVE_ASSET", address(tokens.WrappedNativeToken));
+    }
+
+    function readTokens() public returns (string memory) {
+        string memory root = vm.projectRoot();
+        string memory path = string(
+            abi.encodePacked(root, "/", "contracts/test/foundry/addresses/tokenAddresses.json")
+        );
+        tokensJson = vm.readFile(path);
+        return vm.readFile(path);
+    }
+
+    function getTokens(uint index) public returns (TokenAddresses memory addresses) {
+        readTokens();
+        bytes memory chainTokens = tokensJson.parseRaw(indexChainIds[index]);
+        return abi.decode(chainTokens, (TokenAddresses));
+        //log_named_address("WETH/NATIVE_ASSET", address(tokens.WrappedNativeToken));
+    }
+
+    function createBridgeAdapter(IEtherTokenV06 weth) public returns (IBridgeAdapter bridgeAdapter) {
+        uint chainId;
+
+        assembly {
+            chainId := chainid()
+        }
+        if (chainId == 1) {
+            return IBridgeAdapter(new EthereumBridgeAdapter(weth));
+        } else if (chainId == 56) {
+            return IBridgeAdapter(new BSCBridgeAdapter(weth));
+        } else if (chainId == 137) {
+            return IBridgeAdapter(new PolygonBridgeAdapter(weth));
+        } else if (chainId == 43114) {
+            return IBridgeAdapter(new AvalancheBridgeAdapter(weth));
+        } else if (chainId == 250) {
+            return IBridgeAdapter(new FantomBridgeAdapter(weth));
+        } else if (chainId == 10) {
+            return IBridgeAdapter(new OptimismBridgeAdapter(weth));
+        } else if (chainId == 42161) {
+            return IBridgeAdapter(new ArbitrumBridgeAdapter(weth));
+        }
```

Comment:
I'd suggest you throw if `chainId` is not mapped, otherwise it will be `address(0)` and will confuse the dev

Reply:
good point i didnt consider that